packages/gator-permissions-controller/src/GatorPermissionContoller.test.ts

```diff
@@ -0,0 +1,639 @@
+  PermissionTypes,
+  StoredGatorPermission,
+} from './types';
+
+const MOCK_CHAIN_ID_1: Hex = '0xaa36a7';
+const MOCK_CHAIN_ID_2: Hex = '0x1';
+
+type MockGatorPermissionsStorageEntriesConfig = {
+  [chainId: string]: {
+    nativeTokenStream: number;
+    nativeTokenPeriodic: number;
+    erc20TokenStream: number;
+  };
+};
+
+/**
+ * Creates a mock gator permissions storage entry
+ *
+ * @param amount - The amount of mock gator permissions storage entries to create.
+ * @param mockStorageEntry - The mock gator permissions storage entry to create.
+ * @returns Mock gator permissions storage entry
+ */
+function createMockGatorPermissionsStorageEntries(
+  amount: number,
+  mockStorageEntry: StoredGatorPermission<AccountSigner, PermissionTypes>,
+): StoredGatorPermission<AccountSigner, PermissionTypes>[] {
+  return Array.from({ length: amount }, (_, index: number) => ({
+    ...mockStorageEntry,
+    permissionResponse: {
+      ...mockStorageEntry.permissionResponse,
+      expiry: mockStorageEntry.permissionResponse.expiry + index,
+    },
+  }));
+}
+
+/**
+ * Creates a mock gator permissions storage entry
+ *
+ * @param config - The config for the mock gator permissions storage entries.
+ * @returns Mock gator permissions storage entry
+ */
+function mockGatorPermissionsStorageEntriesFactory(
+  config: MockGatorPermissionsStorageEntriesConfig,
+): StoredGatorPermission<AccountSigner, PermissionTypes>[] {
+  const result: StoredGatorPermission<AccountSigner, PermissionTypes>[] = [];
+
+  // Create entries for each chainId
+  Object.entries(config).forEach(([chainId, counts]) => {
+    const mockNativeTokenStreamStorageEntry: StoredGatorPermission<
+      AccountSigner,
+      NativeTokenStreamPermission
+    > = {
+      permissionResponse: {
+        chainId: chainId as Hex,
+        address: '0xB68c70159E9892DdF5659ec42ff9BD2bbC23e778',
+        expiry: 1750291200,
+        isAdjustmentAllowed: true,
+        signer: {
+          type: 'account',
+          data: { address: '0x4f71DA06987BfeDE90aF0b33E1e3e4ffDCEE7a63' },
+        },
+        permission: {
+          type: 'native-token-stream',
+          data: {
+            maxAmount: '0x22b1c8c1227a0000',
+            initialAmount: '0x6f05b59d3b20000',
+            amountPerSecond: '0x6f05b59d3b20000',
+            startTime: 1747699200,
+            justification:
+              'This is a very important request for streaming allowance for some very important thing',
+          },
+          rules: {},
+        },
+        context: '0x00000000',
+        accountMeta: [
+          {
+            factory: '0x69Aa2f9fe1572F1B640E1bbc512f5c3a734fc77c',
+            factoryData: '0x0000000',
+          },
+        ],
+        signerMeta: {
+          delegationManager: '0xdb9B1e94B5b69Df7e401DDbedE43491141047dB3',
+        },
+      },
+      siteOrigin: 'http://localhost:8000',
+    };
+
+    const mockNativeTokenPeriodicStorageEntry: StoredGatorPermission<
+      AccountSigner,
```

Comment:
nit: I enjoy generic parameters named with a T prefix denoting that they are a generic Type.

It can become harder to grok, especially when it's amongst concrete types.

I'll comment just here, but there's a couple of generic parameters to which this applies.